shell-style.md

```diff
@@ -0,0 +1,347 @@
+# Shell Style
+
+## Global Rules
+
+1. All scripts should pass [Shellcheck][]
+
+   [shellcheck]: https://www.shellcheck.net/
+
+   Add explicit annotations as needed. When adding new scripts, confirm that our
+   CI machinery is configured to run on your files.
+
+1. All scripts end in a `.sh` extension
+
+   This is typically an anti-pattern, but we currently rely on it to locate
+   shell scripts within the source tree for the purposes of Shellcheck on CI.
+
+1. Always use double-quotes, except where interpolation must be prevented
+
+   NOTE: It's acceptable to use single-quotes as a matter of course for script
+   arguments to other interpreters like `sed`, `awk`, and `psql` even if there
+   would be no unintentional interpolations currently.
+
+   **Tip**: quoting a here-doc terminator will prevent interpolation:
+
+   ```sh
+   cat <<EOM
+   Uh-oh, I need to \$escape this
+   EOM
+
+   cat <<'EOM'
+   But not in $this case.
+   EOM
+   ```
+
+## Interpreter
+
+1. Prefer Bash
+
+   We write all of our scripts as Bash. It's available everywhere we need
+   scripts to run and selecting an explicit shell will ensure consistent
+   behavior.
+
+1. Use `#!/usr/bin/env bash`, not `#!/bin/bash`
+
+   This is specifically for OSX folks, where `/bin/bash` is severely out of
+   date. A newer version, installed via Homebrew, may appear first in `$PATH`
+   and should be used when possible.
+
+### Bashisms
+
+Though our scripts are using a Bash shebang, prefer POSIX alternatives where
+possible. This is not about portability, rather it's to reduce to scope of
+understanding required on the part of the reader by preferring simpler,
+more-common, and well-defined constructs.
+
+If you decide to use a Bashism, be consistent within that script. For example,
+if you use `[[` because you want regex or glob matching in one place, don't use
+`[` in other places in the same script.
+
+## Shell Options
+
+1. Use `set -eu`
+
+   `-e` ensures that any unchecked, non-zero exit code halts the script. For
+   cases where a non-zero exit is OK, explicitly append `|| true`.
+
+   `-u` will halt the script if an unset variable is used.
+
+   This avoids bugs like:
+
+   ```sh
+   dir=/tmp/foo
+   rm -rf "$dirr/bar" # typo made this rm -rf /bar
+   ```
+
+2. Avoid `set -x`
+
+   This option outputs every statement before it's executed. Only use this if
+   your script has no function definitions, control flow structures, or
+   conditional expressions; otherwise it is overly noisy.
+
+   ```sh
+   # Bad
+   set -eux
+
+   heyoh() { tar ... }
+
+   if [ this ]; then
+     curl ...
+   fi
+
+   heyoh || mv ...
+
+   # OK
+   set -eux
+
+   curl ...
+   tar ...
+   mv ...
+   ```
+
+## Syntax
+
+1. Use `;` to join `if`, `for` and `while` statements
+
+   ```sh
+   # Bad
+   if
+   then
+   fi
+
+   while
+   do
+   done
+
+   for x
+   do
+   done
+
+   # Good
+   if; then
+   fi
+
+   while; do
+   done
+
+   for x; do
+   done
+   ```
+
+1. Avoid redundant braces:
```

Comment:
I'm not sure I agree with this. I think the point of a lot of good bash style is to avoid the danger of innocuous modifications breaking something in the future. For example, even if you know a variable has not spaces, you quote it when used in case someone adds a space in the future.

Someone could very easily change

```sh
touch "foo/$bar/baz"
```

into 

```sh
touch "foo/$bar_backup/bat
```

Obviously we should be testing changes we make to scripts, but for particularly noisy strings with a variable (especially file paths!) sometimes I like to add the brackets to highlight the interpolation.